packages/omgidl-parser/src/IDLNodes/TypedefIDLNode.ts

```diff
@@ -1,12 +1,7 @@
 import { ReferenceTypeIDLNode } from "./ReferenceTypeIDLNode";
-import { AnyIDLNode, ITypedefIDLNode } from "./interfaces";
+import { ITypedefIDLNode } from "./interfaces";
 import { TypedefASTNode } from "../astTypes";
 
 export class TypedefIDLNode
   extends ReferenceTypeIDLNode<TypedefASTNode>
-  implements ITypedefIDLNode
-{
```

Comment:
why this contructor needed to be changed? It was not being used?

Reply:
All the constructors I removed were because lint was throwing this error: 
`error  Useless constructor  @typescript-eslint/no-useless-constructor`

So after I dug a bit I found out that if the constructor doesn't initialize any properties, accept any parameters, or perform any logic, it can be removed because TypeScript will automatically create a default constructor for the class. And the same applies for when the constructor is simply calling super(). That's why I removed them